core/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -243,11 +272,240 @@ protected AnalyzeResponse shardOperation(AnalyzeRequest request, ShardId shardId
                 IOUtils.closeWhileHandlingException(stream);
             }
         }
+        return tokens;
+    }
 
-        if (closeAnalyzer) {
-            analyzer.close();
+    private DetailAnalyzeResponse detailAnalyze(AnalyzeRequest request, Analyzer analyzer, String field) {
+        DetailAnalyzeResponse detailResponse;
+        final Set<String> includeAttributes = Sets.newHashSet();
+        if (request.attributes() != null && request.attributes().length > 0) {
+            for (String attribute : request.attributes()) {
+                includeAttributes.add(attribute.toLowerCase(Locale.ROOT));
+            }
+        }
+
+        CustomAnalyzer customAnalyzer = null;
+        if (analyzer instanceof CustomAnalyzer) {
+            customAnalyzer = (CustomAnalyzer)analyzer;
+        } else if (analyzer instanceof NamedAnalyzer && ((NamedAnalyzer) analyzer).analyzer() instanceof CustomAnalyzer) {
+            customAnalyzer = (CustomAnalyzer) ((NamedAnalyzer) analyzer).analyzer();
+        }
+
+        if (customAnalyzer != null) {
+            // customAnalyzer = divide charfilter, tokenizer tokenfilters
+            CharFilterFactory[] charfilterFactories = customAnalyzer.charFilters();
+            TokenizerFactory tokenizerFactory = customAnalyzer.tokenizerFactory();
+            TokenFilterFactory[] tokenfilterFactories = customAnalyzer.tokenFilters();
+
+            Map<String, List<String>> charFiltersTexts = Maps.newHashMapWithExpectedSize(charfilterFactories.length);
+            Map<String, TokenListCreator> tokenFiltersTokenListCreator = Maps.newHashMapWithExpectedSize(tokenfilterFactories.length);
+
+            TokenListCreator tokenizerTokenListCreator = new TokenListCreator();
+
+            for (String text : request.text()) {
+                String charFilteredSource = text;
+
+                Reader reader = new FastStringReader(text);
+                if (charfilterFactories != null) {
```

Comment:
Here you test that charfilterFactories is null. But you wrote before `Maps.newHashMapWithExpectedSize(charfilterFactories.length);`

So I suppose that charfilterFactories can not be really null?